### PR TITLE
Use Url and rewrite domain matching

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by jdm** [comment]
Eh?

**Comment by nox** [comment]
No idea what happened.

File: src/repr.rs [hunk 1]
```diff
@@ -56,38 +58,46 @@ pub enum LoadType {
...
-            if request.url.find(&format!(".{}", domain)).is_some() {
+        for candidate in &*self.exact {
+            if domain == candidate {
                 return true;
             }
         }
-
+        for suffix in &*self.subdomain {
+            match domain.len().cmp(&suffix.len()) {
+                Ordering::Equal if domain == suffix => return true,
```
Comment (jdm) [inline src/repr.rs:79]: Let's add a test for a subdomain match against the main domain, since clearly that didn't used to work.

**Comment by jdm** [comment]
Looks good with the addition of a new test!

**Comment by nox** [comment]
I've added a test and I've made the tests use the same convenience functions as the parser, to test more of the library.

**Comment by jdm** [comment]
Thanks!

**Pull request merged**